check_source.py

```diff
@@ -187,8 +193,60 @@ def check_source_submission(self, source_project, source_package, source_revisio
+            # if there are any add a review for the security team
+            # maybe add the found warnings to the message for the review
+            self.add_review(self.request, by_group='security-team', msg="The package is submitted to an official product and it has warnings that indicate that it need to go through a security review. Those warnings can only be ignored in devel projects. For more information please read: {}.".format(CheckSource.AUDIT_BUG_URL))
+        if self.suppresses_whitelist_warnings( source_project, source_package ):
+            self.add_review(self.request, by_group='security-team', msg="The package is submitted to an official product and it has warnings that indicate that it need to go through a security review. Those warnings can only be ignored in devel projects. For more information please read: {}.".format(CheckSource.AUDIT_BUG_URL))
+
         return True
 
+    def suppresses_whitelist_warnings( self, source_project, source_package):
+        # checks if there's a rpmlintrc that suppresses warnings that we check
+        url = osc.core.makeurl(self.apiurl, ['source', source_project, source_package, source_package+'-rpmlintrc'])
+        found_entries = []
+        try:
+            result = osc.core.http_GET(url)
+            contents = str(result.read())
+            for entry in self.bad_rpmlint_entries:
+                if re.search('addFilter\(["\']'+entry,contents):
+                    self.logger.info(f'found suppressed whitelist warning {entry}')
+                    found_entries.append(entry)
+            return found_entries
+        except HTTPError as e:
+            # probably doesn't have a rpmlintrc
+            pass
+        return False
+
+    def has_whitelist_warnings( self, source_project, source_package, target_project, target_package ):
+        # this checks if this is a submit to an product project and it has warnings for non-whitelisted permissions/files
+        found_entries = []
+        # we only want this check in official projects
+        if target_project.startswith('openSUSE:') or target_project.startswith('SUSE:'):
+            url = osc.core.makeurl(self.apiurl, ['build', target_project])
+            xml = ET.parse(osc.core.http_GET(url)).getroot()
+            for f in xml.findall('entry'):
+                # we check all repos in the source project for error sthat exist in the target project
+                repo = f.attrib['name']
+                query = { 'last' : 1, }
+                # FIXME: Hardcoded architecture?
+                url = osc.core.makeurl(self.apiurl, ['build', source_project, repo, 'x86_64', source_package,'_log'], query = query)
+                try:
+                    result = osc.core.http_GET(url)
+                    contents = str(result.read())
+                    for entry in self.bad_rpmlint_entries:
+                        # FIXME change that to W once we change it to a warning
```

Comment:
As in you will change to warning once this tool is deployed? If so then just change it in this code now and once merged it is auto-deployed and it can be changed in rpmlint.

Reply:
I had it in there to facilitate testing, will change it now. You would need to change it back if you want to test it with the submit mentioned below